Sources_v3/StreamChatUI/Common Views/CurrentChatUserAvatarView.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class CurrentChatUserAvatarView<ExtraData: UIExtraDataTypes>: AvatarView, AppearanceSetting {
+    // MARK: - Content
+    
+    public var currentUser: _CurrentChatUser<ExtraData.User>? {
+        didSet { updateContent() }
+    }
+    
+    open func updateContent() {
+        if let imageURL = currentUser?.imageURL {
+            imageView.setImage(from: imageURL)
+        } else {
+            imageView.image = nil
+        }
+    }
+    
+    // MARK: - Overrides
+    
+    override open var intrinsicContentSize: CGSize {
+        .init(width: 28, height: 28)
+    }
+    
+    override open func setupAppearance() {
+        applyDefaultAppearance()
+        
+        super.setupAppearance()
```

Comment:
Isn't this called on `super.init()`?

Reply:
That's the first limitation we face with `AppearanceSetting`: if we subclass a view which implements `AppearanceSetting`, the base view method is invoked instead of a one implemented in subclass